File: app/src/test/java/com/nytimes/android/external/register/di/GsonFactory.java
package com.nytimes.android.external.register.di;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.nytimes.android.external.register.model.GsonAdaptersConfig;
import com.nytimes.android.external.register.model.GsonAdaptersConfigSku;

public final class GsonFactory {

    private GsonFactory() {
        // utility
    }

    public static Gson create() {
        return new GsonBuilder()
                .registerTypeAdapterFactory(new GsonAdaptersConfig())
                .registerTypeAdapterFactory(new GsonAdaptersConfigSku())
                .create();
    }
}

File: app/src/test/java/com/nytimes/android/external/register/di/GsonFactory.kt
package com.nytimes.android.external.register.di

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.nytimes.android.external.register.model.GsonAdaptersConfig
import com.nytimes.android.external.register.model.GsonAdaptersConfigSku

object GsonFactory {

    @JvmStatic
    fun create(): Gson {
        return GsonBuilder()
                .registerTypeAdapterFactory(GsonAdaptersConfig())
                .registerTypeAdapterFactory(GsonAdaptersConfigSku())
                .create()
    }
}
